include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
+                          const LevelInd &cell);
 
         /**
-         * Get the deal.II Triangulation associated with this particle world.
-         *
-         * @return const pointer to associated Triangulation
+         * Initialize the particle properties.
          */
-        const parallel::distributed::Triangulation<dim> *get_triangulation()
-        {
-          return triangulation;
-        };
+        void initialize_particles();
 
         /**
-         * Get the deal.II DoFHandler associated with this particle world.
-         *
-         * @return The DoFHandler for this world.
+         * Update the particle properties if necessary.
          */
-        const DoFHandler<dim> *get_dof_handler() const
-        {
-          return dof_handler;
-        };
+        void update_particles();
 
         /**
-         * Set the deal.II DoFHandler associated with this particle world.
-         *
-         * @param [in] new_dh The new DoFHandler for this world.
+         * Advect the particle positions by one integration step. Needs to be
+         * called until integrator->continue() returns false.
          */
-        void set_dof_handler(const DoFHandler<dim> *new_dh)
-        {
-          dof_handler = new_dh;
-        };
+        void advect_particles();
 
         /**
-         * Get the deal.II BlockVector solution associated with this particle
-         * world.
-         *
-         * @return The BlockVector solution for this world.
+         * Access to particles in this world.
          */
-        const LinearAlgebra::BlockVector *get_solution() const
-        {
-          return solution;
-        };
+        std::multimap<LevelInd, Particle<dim> > &
+        get_particles();
 
         /**
-         * Set the deal.II BlockVector solution associated with this particle
-         * world.
-         *
-         * @param [in] new_solution The new LinearAlgebra::BlockVector
-         * solution for this world.
+         * Const access to particles in this world.
          */
-        void set_solution(const LinearAlgebra::BlockVector *new_solution)
-        {
-          solution = new_solution;
-        };
+        const std::multimap<LevelInd, Particle<dim> > &
+        get_particles() const;
 
         /**
-         * Set the particle Integrator scheme for this particle world.
-         *
-         * @param [in] new_integrator The new Integrator scheme for this
-         * world.
+         * Advance particles by the old timestep using the current
+         * integration scheme. This accounts for the fact that the tracers
+         * are actually still at their old positions and the current timestep
+         * length is already updated for the next step at the time this
+         * function is called.
          */
-        void set_integrator(Integrator::Interface<dim, T> *new_integrator)
-        {
-          integrator = new_integrator;
-        };
+        void advance_timestep();
```

Comment:
Does this function call `advect_particles()`? If so, could `advect_particles()` be made `private`?